Common/src/Storage/TaskCreationRequest.cs

```diff
@@ -19,6 +19,8 @@
 
 namespace ArmoniK.Core.Common.Storage;
 
-public record TaskRequest(string              Id,
-                          IEnumerable<string> ExpectedOutputKeys,
-                          IEnumerable<string> DataDependencies);
+public record TaskCreationRequest(string              TaskId,
+                                  string              PayloadId,
+                                  TaskOptions         Options,
```

Comment:
Not here, by design, we intend to put task options in the task table that come from the session if null during submission. As we check the options before creating this request, it does not make sense to make it nullable. On the opposite, It will make it less clear.
Refactor options management is not a goal of this development and can be done in another PR.

Reply:
Good call. I missed that the merge with the session TaskOption was already done at this point in time.